src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -113,8 +116,17 @@ public void addPerson(Person p) {
         persons.add(p);
     }
 
-    /** Adds a job to the address book. */
+    /**
+     * Adds a job to the address book.
+     * The existence of the company referenced by the job creation is checked here.
+     */
     public void addJob(Job j) {
+        boolean companyExists = StreamSupport
+                .stream(companies.spliterator(), false)
+                .anyMatch(x -> j.getCompany().matchesCompanyName(x.getName()));
```

Comment:
Efficiency related -> j.getCompany() can be computed once and saved to a variable.

Reply:
Ok I can resolve that

Reply:
Hahhaa its optional, just a small nit.